CMakeLists.txt

```diff
@@ -44,36 +47,41 @@ set(CMAKE_RUNTIME_OUTPUT_DIRECTORY ${CMAKE_BINARY_DIR}/bin)
 file(GLOB SOURCES "edlib/src/*.cpp")
 
 # Create libraries.
-add_library(edlib SHARED ${SOURCES})
+add_library(edlib ${SOURCES})
+set_target_properties(edlib
+    PROPERTIES
+    VERSION ${PROJECT_VERSION}
+    SOVERSION ${PROJECT_VERSION})
 
 target_include_directories(edlib PUBLIC
     $<BUILD_INTERFACE:${CMAKE_CURRENT_SOURCE_DIR}/edlib/include>
     $<INSTALL_INTERFACE:include>)
 
-add_library(edlib_static STATIC ${SOURCES})
-
-target_include_directories(edlib_static PUBLIC
-    $<BUILD_INTERFACE:${CMAKE_CURRENT_SOURCE_DIR}/edlib/include>
-    $<INSTALL_INTERFACE:include>)
-
 # Build binaries.
 add_executable(helloWorld apps/hello-world/helloWorld.c)
-target_link_libraries(helloWorld edlib_static)
+target_link_libraries(helloWorld edlib)
 
 include(CTest)
 if (BUILD_TESTING)
   add_executable(runTests test/runTests.cpp)
-  target_link_libraries(runTests edlib_static)
+  target_link_libraries(runTests edlib)
 
   add_test(edlib_tests ${CMAKE_RUNTIME_OUTPUT_DIRECTORY}/runTests)
 endif()
 
 if (NOT WIN32) # If on windows, do not build binaries that do not support windows.
   add_executable(edlib-aligner apps/aligner/aligner.cpp)
-  target_link_libraries(edlib-aligner edlib_static)
+  target_link_libraries(edlib-aligner edlib)
 endif()
 
 
 # Create target 'install' for installing libraries.
-install(TARGETS edlib edlib_static DESTINATION ${CMAKE_INSTALL_LIBDIR})
+install(TARGETS edlib DESTINATION ${CMAKE_INSTALL_LIBDIR})
 install(FILES edlib/include/edlib.h DESTINATION ${CMAKE_INSTALL_INCLUDEDIR})
+
+# configure and install pkg-config file
+configure_file(
+    ${CMAKE_CURRENT_SOURCE_DIR}/edlib.pc.in
+    ${CMAKE_CURRENT_BINARY_DIR}/edlib-1.pc
```

Comment:
Ok this is cool :)! Why edlib-1 (why number 1)?

Reply:
In general, versioning the API is always a good thing. Say you ever made a new edlib-2 that was API incompatible, then without versioning the API, they wouldn't be side-by-side installable. See also:

https://tecnocode.co.uk/2014/12/09/a-checklist-for-writing-pkg-config-files/#fn1-869
https://developer.gnome.org/programming-guidelines/unstable/parallel-installability.html.en#solution

Reply:
Aha very good, makes sense! Same like python-2 and python-3. Basically, we use the major version number since that is the one that indicates API breaking changes.
Ok cool, thanks!